self-run-agents/instances/azp-build-asg/init.sh.tpl

```diff
@@ -6,66 +6,30 @@ echo "$instance_id" > /etc/hostname
 sudo -u azure-pipelines /bin/bash -c 'cd /srv/azure-pipelines && ./config.sh --unattended --acceptteeeula --url https://dev.azure.com/cncf/ --pool ${azp_pool_name} --token ${azp_token}'
-# The Service Name Generated Contains hostname information, which we don't want. Rename.
-sudo -u azure-pipelines /bin/bash -c 'cd /srv/azure-pipelines && sed -i "s,SVC_NAME=.*,SVC_NAME=azp-agent.service,g" ./svc.sh'
-
-mkdir -p /usr/local/bin
-echo '#!/usr/bin/env bash
-
-mkdir -p /run/aws-protection-data/
-
-role_name=$(wget -q -O - http://169.254.169.254/latest/meta-data/iam/security-credentials)
-wget -q -O - "http://169.254.169.254/latest/meta-data/iam/security-credentials/$role_name" > /run/aws-protection-data/creds.json
-wget -q -O - http://169.254.169.254/latest/dynamic/instance-identity/document > /run/aws-protection-data/iid.json
-echo -n "${asg_name}" > /run/aws-protection-data/asg-name
-
-chmod 0400 /run/aws-protection-data/creds.json
-chmod 0400 /run/aws-protection-data/iid.json
-chmod 0400 /run/aws-protection-data/asg-name
-chown azure-pipelines:azure-pipelines /run/aws-protection-data/creds.json
-chown azure-pipelines:azure-pipelines /run/aws-protection-data/iid.json
-chown azure-pipelines:azure-pipelines /run/aws-protection-data/asg-name
-' > /usr/local/bin/aws-metadata-refresh.sh
-chmod +x /usr/local/bin/aws-metadata-refresh.sh
-
-echo "[Unit]
-Description=Refersh User Data Credentials for AZP
-Wants=aws-metadata-refresh.timer
-
-[Service]
-ExecStart=/usr/local/bin/aws-metadata-refresh.sh
-User=root
-
-[Install]
-WantedBy=multi-user.target" > /lib/systemd/system/aws-metadata-refresh.service
-echo "[Unit]
-Description=Renew AWS Creds once every half hour
-Requires=aws-metadata-refresh.service
-
-[Timer]
-Unit=aws-metadata-refresh.service
-OnUnitInactiveSec=30m
-RandomizedDelaySec=30m
-
-[Install]
-WantedBy=timers.target" > /lib/systemd/system/aws-metadata-refresh.timer
-systemctl daemon-reload
-systemctl enable aws-metadata-refresh.service aws-metadata-refresh.timer
-systemctl start aws-metadata-refresh.service aws-metadata-refresh.timer
-/usr/local/bin/aws-metadata-refresh.sh
+sudo -u azure-pipelines mkdir /srv/azure-pipelines/_work
 
 # Block Azure Pipelines from Making Requests to Local Instance User Data.
-iface=$(ls /sys/class/net/ | grep -v lo | grep -v docker0)
-iptables -A OUTPUT -o "$iface" -s 169.254.169.254 -m owner --uid-owner azure-pipelines -j DROP
-iptables -A OUTPUT -o "$iface" -d 169.254.169.254 -m owner --uid-owner azure-pipelines -j DROP
+iptables -A OUTPUT -d 169.254.169.254 -m owner ! --uid-owner root -j DROP
+iptables -A OUTPUT -s 169.254.169.254 -m owner ! --uid-owner root -j DROP
 iptables -I DOCKER-USER -s 169.254.169.254 -j DROP
 iptables -I DOCKER-USER -d 169.254.169.254 -j DROP
 
 # Start AZP Agent.
-(cd /srv/azure-pipelines && ./svc.sh install azure-pipelines)
-systemctl daemon-reload
-systemctl enable azp-agent.service
-systemctl start azp-agent.service
\ No newline at end of file
+function terminate {
```

Comment:
This leaves the token locally running where it can be reached by a user. We can also expand the lifecycle hook for the lambda so it terminates things.

Reply:
Which tokens? The permission of tokens in `/run` didn't change

Reply:
removed

Reply:
Any reason it has to be a minute? Couldn't we just shut it down immediately?

Reply:
It is just to leave myself (and on-call in future) some room to login via SSH to debug in some situation. `shutdown -h now` doesn't leave space to cancel it.